src/backends/backend/python/python_interpreter.cc

```diff
@@ -0,0 +1,902 @@
+
+  TRITONSERVER_Error* err = model_config.Parse(buffer, byte_size);
+  RETURN_IF_ERROR(TRITONSERVER_MessageDelete(config_message));
+  RETURN_IF_ERROR(err);
+
+  *state = new ModelInstanceState(
+      model_state, triton_model_instance, instance_name, instance_kind,
+      instance_id, std::move(model_config));
+
+  return nullptr;
+}
+
+ModelInstanceState::~ModelInstanceState()
+{
+  // Close python interpreter.
+  grpc::ClientContext context;
+  ni::StatusCode result;
+  ni::Empty null_msg;
+
+  const auto err = stub->InterpreterShutdown(&context, null_msg, &result);
+  if (!err.ok() || result.code() != 0) {
+    LOG_ERROR << "Cannot shutdown interpreter gracefully: "
+              << err.error_message() << std::endl;
+  }
+
+  int status;
+  waitpid(interpreter_pid_, &status, 0);
+  unlink(domain_socket_.substr(7).c_str());
+}
+
+int
+ModelInstanceState::GetInputTensor(
+    const uint32_t iidx, TRITONBACKEND_Request* request,
+    ni::InferenceData* input_tensor,
+    std::vector<TRITONBACKEND_Response*>& responses, size_t r,
+    uint32_t& batch_size)
+{
+  const char* input_name;
+  // Load iidx'th input name
+  RESPOND_AND_RETURN_IF_ERROR(
+      request, TRITONBACKEND_RequestInputName(request, iidx, &input_name));
+
+  // Load iidx'th input
+  TRITONBACKEND_Input* in;
+  RESPOND_AND_RETURN_IF_ERROR(
+      request, TRITONBACKEND_RequestInput(request, input_name, &in));
+
+  // Load input properties
+  TRITONSERVER_DataType input_dtype;
+  const int64_t* input_shape;
+  uint32_t input_dims_count;
+  uint64_t input_byte_size;
+  uint32_t input_buffer_count;
+  GUARDED_RESPOND_IF_ERROR(
+      responses, r,
+      TRITONBACKEND_InputProperties(
+          in, &input_name, &input_dtype, &input_shape, &input_dims_count,
+          &input_byte_size, &input_buffer_count));
+
+  if (iidx == 0) {
+    batch_size = input_shape[0];
+  }
+
+  // Update input_tensor.
+  input_tensor->set_name(input_name);
+  input_tensor->set_dtype(static_cast<inference::DataType>(input_dtype));
+
+  for (size_t j = 0; j < input_dims_count; ++j) {
+    input_tensor->add_dims(input_shape[j]);
+  }
+
+  // Load raw data into input_tensor raw data.
+  std::string* data_buffer = input_tensor->mutable_raw_data();
+  const void* input_buffer = nullptr;
+  uint64_t buffer_byte_size = 0;
+  TRITONSERVER_MemoryType input_memory_type = TRITONSERVER_MEMORY_CPU;
+  int64_t input_memory_type_id = 0;
+  for (size_t j = 0; j < input_buffer_count; ++j) {
+    GUARDED_RESPOND_IF_ERROR(
+        responses, r,
+        TRITONBACKEND_InputBuffer(
+            in, j, &input_buffer, &buffer_byte_size, &input_memory_type,
+            &input_memory_type_id));
+    data_buffer->append((const char*)input_buffer, buffer_byte_size);
+  }
+
+  return nic::ErrorCodes::Success;
+}
+
+TRITONSERVER_Error*
+ModelState::Create(TRITONBACKEND_Model* triton_model, ModelState** state)
+{
+  TRITONSERVER_Message* config_message;
+  // TODO: Make this work with every version
```

Comment:
Remove TODO. You are requesting "version 1" format of the configuration, and that is what Triton will give you.